src/main.ts

```diff
@@ -0,0 +1,128 @@
+            case 'push':
+                account = event_payload_data['pusher']['name'];
+                message = `**Commit to GitHub** by ${account}`;
+                url = event_payload_data['compare'];
+                details = `Comment: ${event_payload_data['head_commit']['message']}`;
+                break;
+            case 'pull_request':
+                message = `**PR submitted to Github**: ${event_payload_data['pull_request']['title']}`;
+                url = event_payload_data['pull_request']['html_url'];
+                details = `Pr for merging ref ${event_payload_data['pull_request']['head']['ref']} to base branch ${event_payload_data['base']['ref']}`;
+                break;
+            case 'issue':
+                message = `**New/updated GitHub issue**: ${event_payload_data['issue']['title']}`;
+                url = event_payload_data['issue']['html_url'];
+                details = `Issue state: ${event_payload_data['issue']['state']}  - assignee: ${event_payload_data['issue']['assignee']}`;
+                break;
+            case 'issue_comment':
+                message = `**A Github issue comment was posted**: ${event_payload_data['comment']['body']}`;
+                url = event_payload_data['issue']['html_url'];
+                details = `Issue state: ${event_payload_data['issue']['state']} - assignee: ${event_payload_data['issue']['assignee']}`;
+                break;
+            default:
+                if (event_name) {
+                    message = event_name.replace(/_/g, ".");
+                } else {
+                    message = "A GitHub Actions event has occurred"
+                }
+                url = `https://github.com/${repository}/actions`;
+        }
+
+        if (!details) {
+            details = core.getInput('details');
+        }
+    } catch (error) {
+        core.setFailed(error.message);
+    }
+}
+
+
+async function notifyTeams() {
+    const matches = msteams_webhook_url.match(/^https?\:\/\/([^\/?#]+)(.*)/i);
+    const hostname_match = matches && matches[1];
+    const path_match = matches && matches[2];
+
+    let req_data = JSON.stringify(require(jsonPath));
+    req_data  = req_data.replace(/GITHUB_WORKFLOW/g, `${workflow}`)
+                        .replace(/GITHUB_REPOSITORY/g, `${repository}`)
+                        .replace(/GITHUB_REF/g, `${branch}`)
+                        .replace(/GITHUB_TRIGGER_EVENT_DETAILS/g, `${details}`)
+                        .replace(/GITHUB_TRIGGER_EVENT/g, `${message}`)
+                        .replace(/GITHUB_EVENT_URL/g, `${url}`)
+                        .replace(/GITHUB_STATUS/g, `${job_status}`);
+
+    const options = {
+        hostname: `${hostname_match}`,
+        port: 443,
+        path: `${path_match}`,
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+            'Content-Length': req_data.length
+        }
+    };
+
+    const req = https.request(options, (res) => {
```

Comment:
consider using something like `node-fetch`. whilst it's not a native node module, it's a bit easier to deal with than the `https` module.

Reply:
Really good hint, I was between http and request before. If that is ok with you I will keep it as a backup, since using http works without issues at the current point - we are anyway not doing anything too fancy here, just sending the payload to the webhook directly (one call per action)